packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -0,0 +1,526 @@
+    }
+  }
+
+  void _handleTap() {
+    if (!_isSelectMode)
+      return;
+    if (_lastPointerLocation != null) {
+      _inspectAt(_lastPointerLocation);
+
+      if (_selection != null) {
+        // Notify debuggers to open an inspector on the object.
+        developer.inspect(_selection.current);
+        print(_selection.current.toStringDeep());
+      }
+    }
+    setState(() {
+      _isSelectMode = false;
+    });
+  }
+
+  void _handleEnableSelect() {
+    setState(() {
+      _isSelectMode = true;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> children = <Widget>[];
+    children.add(new GestureDetector(
+      onTap: _handleTap,
+      onPanDown: _handlePanDown,
+      onPanEnd: _handlePanEnd,
+      onPanUpdate: _handlePanUpdate,
+      behavior: HitTestBehavior.opaque,
+      excludeFromSemantics: true,
+      child: new IgnorePointer(
+        ignoring: _isSelectMode,
+        key: _ignorePointerKey,
+        ignoringSemantics: false,
+        child: widget.child,
+      ),
+    ));
+    if (!_isSelectMode) {
+      children.add(new Positioned(
+        left: _kInspectButtonMargin,
+        bottom: _kInspectButtonMargin,
+        child: new FloatingActionButton(
+          child: const Icon(Icons.search),
+          onPressed: _handleEnableSelect,
+          mini: true,
+        ),
+      ));
+    }
+    children.add(new _InspectorOverlay(selection: _selection));
+    return new Stack(children: children);
+  }
+}
+
+/// Mutable selection state of the inspector.
+class _InspectorSelection {
+  List<RenderObject> _candidates = <RenderObject>[];
+
+  /// Render objects that are candidates to be selected.
+  ///
+  /// Tools may wish to iterate through the list of candidates
+  List<RenderObject> get candidates => _candidates;
+
+  /// Index within the list of candidates that is currently selected.
+  int index = 0;
+
+  set candidates(List<RenderObject> value) {
+    _candidates = value;
+    index = 0;
+  }
+
+  void clear() {
+    _candidates = <RenderObject>[];
+    index = 0;
+  }
+
+  RenderObject get current {
+    return candidates != null && index < candidates.length ? candidates[index] : null;
+  }
+
+  bool get active => current != null && current.attached;
+}
+
+class _InspectorOverlay extends LeafRenderObjectWidget {
+  const _InspectorOverlay({
+    Key key,
+    @required this.selection,
+  }) : super(key: key);
+
+  final _InspectorSelection selection;
+
+  @override
+  _RenderInspectorOverlay createRenderObject(BuildContext context) {
+    return new _RenderInspectorOverlay(selection: selection);
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, _RenderInspectorOverlay renderObject) {
+    renderObject.selection = selection;
+  }
+}
+
+class _RenderInspectorOverlay extends RenderBox {
+  /// The arguments must not be null.
+  _RenderInspectorOverlay({ @required this.selection }) : assert(selection != null);
+
+  _InspectorSelection selection;
```

Comment:
Oh I see. Sorry, I wasn't clear. The markNeedsPaint call should be on the render object, not the render object widget. The "selection" property should be a getter/setter pair and the setter should call markNeedsPaint if the new value is different than the old. See other render objects for examples of the pattern.